src/java/org/apache/cassandra/cql3/statements/schema/AlterTableStatement.java

```diff
@@ -391,6 +405,15 @@ private AlterOptions(String keyspaceName, String tableName, TableAttributes attr
             this.attrs = attrs;
         }
 
+        @Override
+        public void validate(ClientState state)
+        {
+            super.validate(state);
+
+            Guardrails.disallowedTableProperties.ensureAllowed(attrs.updatedProperties(), state);
+            Guardrails.ignoredTableProperties.maybeIgnoreAndWarn(attrs.updatedProperties(), attrs::removeProperty, state);
```

Comment:
do we really need to add warn/fail call sites?  Shouldn't this be

```
// this performs both warn and failure checks
Guardrails.allowTableProperties.check(attrs.updatedProperties(), attrs::removeProperty, state);
```

Reply:
Makes sense, I have merged both types of guardrail into a single one.